turbinia/workers/__init__.py

```diff
@@ -413,64 +414,64 @@ def run_wrapper(self, evidence):
     Returns:
       A TurbiniaTaskResult object
     """
-    # TODO(aarontp): Add locking: https://github.com/google/turbinia/issues/221
-    log.info('Starting Task {0:s} {1:s}'.format(self.name, self.id))
-    original_result_id = None
-    try:
-      self.result = self.setup(evidence)
-      original_result_id = self.result.id
-      self._evidence_config = evidence.config
-      self.result = self.run(evidence, self.result)
-    # pylint: disable=broad-except
-    except (TurbiniaException, Exception) as e:
-      msg = '{0:s} Task failed with exception: [{1:s}]'.format(
-          self.name, str(e))
-      log.error(msg)
-      log.error(traceback.format_exc())
-      if self.result:
-        self.result.log(msg)
-        self.result.log(traceback.format_exc())
-        if hasattr(e, 'message'):
-          self.result.set_error(e.message, traceback.format_exc())
-        else:
-          self.result.set_error(e.__class__, traceback.format_exc())
-        self.result.status = msg
-      else:
-        log.error('No TurbiniaTaskResult object found after task execution.')
-
-    self.result = self.validate_result(self.result)
-
-    # Trying to close the result if possible so that we clean up what we can.
-    # This has a higher likelihood of failing because something must have gone
-    # wrong as the Task should have already closed this.
-    if self.result and not self.result.closed:
-      msg = 'Trying last ditch attempt to close result'
-      log.warning(msg)
-      self.result.log(msg)
-
-      if self.result.status:
-        status = self.result.status
-      else:
-        status = 'No previous status'
-      msg = ('Task Result was auto-closed from task executor on {0:s} likely '
-             'due to previous failures.  Previous status: [{1:s}]'.format(
-                 self.result.worker_name, status))
-      self.result.log(msg)
+    with filelock.FileLock(config.LOCK_FILE):
```

Comment:
It seemed to work during manual testing (I wasn't actually sure what a good way to test it with unit tests was without doing anything funny with subprocesses or threading).  In the examples in the README it shows this pattern though:
https://github.com/benediktschmitt/py-filelock#examples

Reply:
FWIW, what I did for manual testing was I added a sleep(60) to the Stat task, and had the Job create three tasks.  After starting a single worker, it took ~3min to execute all tasks (instead of ~1min), and it also logs something when it hits the lock.

Reply:
OK, I debugged this a bit, and this works as expected - the lock is acquired and released in the context manager.